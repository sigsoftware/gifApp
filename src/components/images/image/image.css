.image {
    width: 250px;
    height: 150px;
    cursor: pointer;
    padding-right: 45px;
    border-radius: 6px;
}


@media (max-width:560px) {
    .image {
        width: 250px;
        height: 150px;
        cursor: pointer;
        padding-right: 45px;
        border-radius: 6px;
    }
}