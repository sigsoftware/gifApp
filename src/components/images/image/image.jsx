import React from 'react'
import "./image.css";
import PropTypes from "prop-types";

const Image = ({ image }) => {
    return (
        <>
            <img src={image?.imageUrl} className="image" alt={image?.title} />
        </>
    )
}

Image.proTypes = {
    image: PropTypes.object.isRequired,
}

Image.defaultProps = {
    image: {},
}

export { Image };

