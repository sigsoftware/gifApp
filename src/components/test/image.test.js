import { Image } from "./../images/image/image";
import { shallow } from "enzyme";

describe("Pruebas a realizar al componente de imagen ", () => {

    let wrapper;

    let image = {
        id: 1,
        imageUrl: "https://www.infobae.com/new-resizer/nLv3x5W1j7DasOsVz_oq_YLetWs=/1200x628/filters:format(webp):quality(85)//arc-anglerfish-arc2-prod-infobae.s3.amazonaws.com/public/EUURIT3YORELNJGPEDTTKBXOXU",
        title: "test",
    }

    

    beforeEach(() => {
        wrapper = shallow(<Image image={image} />);
    })

    test("Testear mediante un snapshot el componente de imagen", () => {
        expect(wrapper).toMatchSnapshot();
    })

    test("Testear que estoy recibiendo como alt de la imagen el texto test", async () => {
        let imageTarget = wrapper.find('img').props();
        expect(imageTarget.alt).toBe(image.title);
    })

    test("Testear que la direccion es la impuesta en el objeto image", async () => {
        let imageTarget = wrapper.find('img').props();
        expect(imageTarget.src).toBe(image.imageUrl);
    })
})