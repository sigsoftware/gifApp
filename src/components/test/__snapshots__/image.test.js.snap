// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a realizar al componente de imagen  Testear mediante un snapshot el componente de imagen 1`] = `
<Fragment>
  <img
    alt="test"
    className="image"
    src="https://www.infobae.com/new-resizer/nLv3x5W1j7DasOsVz_oq_YLetWs=/1200x628/filters:format(webp):quality(85)//arc-anglerfish-arc2-prod-infobae.s3.amazonaws.com/public/EUURIT3YORELNJGPEDTTKBXOXU"
  />
</Fragment>
`;
